PowerEditor/src/WinControls/ReadDirectoryChanges/ReadDirectoryChangesPrivate.cpp

```diff
@@ -149,7 +149,7 @@ void CReadChangesRequest::ProcessNotification()
 
 		CStringW wstrFilename(fni.FileName, fni.FileNameLength/sizeof(wchar_t));
 		// Handle a trailing backslash, such as for a root directory.
-		if (wstrFilename.Right(1) != L"\\")
+		if (m_wstrDirectory.Right(1) != L"\\")

```

Comment:
Is this modification intended to fix any bug?

Reply:
Actually no bug has been seen so far, during debugging this piece of code, I came to know that this part should be modified. Refer debugger info as below for currently being monitored file (D:\Test\Test.txt)-

![image](https://user-images.githubusercontent.com/14791461/36935288-2741fc16-1f1c-11e8-9c47-8707c5503508.png)

![image](https://user-images.githubusercontent.com/14791461/36935332-af247a32-1f1c-11e8-8ac6-3b535dc3e2fd.png)

Reply:
Also, as per existing code, refer the below screenshot 1 and with updated code screenshot 2.

**With existing code:**
![image](https://user-images.githubusercontent.com/14791461/36946280-7acbff84-1fe0-11e8-894a-4bd634e11abc.png)

**With Updated code:**
![image](https://user-images.githubusercontent.com/14791461/36946282-85912976-1fe0-11e8-8196-57408804b22b.png)

Reply:
So lines 89 to 93 of `NppIO.cpp` can be removed as their intended use (i.e. fix problem of monitoring files which are in root) are now (correctly) solved with this patch?

Reply:
@MAPJe71, Yes, these lines can be removed. But there is no harm even if we keep those.

Reply:
Might be wise to add some (extra) comment to these lines to prevent questions/confusion in the future.